examples/sphere/shallow_water.jl

```diff
@@ -416,106 +435,147 @@ if haskey(ENV, "CI_PERF_SKIP_RUN") # for performance analysis
-        relL1err[t] = norm(sol.u[t].h .- Y.h, 1) / norm(Y.h, 1)
+if usempi
+    walltime = @elapsed sol = OrdinaryDiffEq.solve!(integrator)
+    ClimaComms.iamroot(context) && println("walltime = $walltime (sec)")
+else
+    sol = @timev OrdinaryDiffEq.solve!(integrator)
+end
+sol_global = []
+if usempi
+    for sol_step in sol.u
+        sol_step_values_global =
+            DataLayouts.gather(context, Fields.field_values(sol_step))
+        if ClimaComms.iamroot(context)
+            sol_step_global = Fields.Field(sol_step_values_global, global_space)
+            push!(sol_global, sol_step_global)
+        end
     end
-    Plots.png(
-        Plots.plot(
-            [1:dt:T],
-            relL1err,
-            xlabel = "time [s]",
-            ylabel = "Relative L₁ err",
-            label = "",
-        ),
-        joinpath(path, "HeightRelL1errorVstime.png"),
-    )
-    linkfig(
-        relpath(
-            joinpath(path, "HeightRelL1errorVstime.png"),
-            joinpath(@__DIR__, "../.."),
-        ),
-        "Height relative L1 error over time",
-    )
-
-    relL2err = Array{Float64}(undef, div(T, dt))
-    for t in 1:div(T, dt)
-        relL2err[t] = norm(sol.u[t].h .- Y.h) / norm(Y.h)
+end
+solution = usempi ? sol_global : sol.u
+if !usempi || ClimaComms.iamroot(context)
+    # Plot variables and auxiliary function
+    ENV["GKSwstype"] = "nul"
+    using ClimaCorePlots, Plots
+    Plots.GRBackend()
+    dir = "cg_sphere_shallowwater_$(test_name)"
+    dir = "$(dir)_$(test_angle_name)"
+    path = joinpath(@__DIR__, "output", dir)
+    mkpath(path)
+
+    function linkfig(figpath, alt = "")
+        # Buildkite-agent upload figpath
+        # Link figure in logs if we are running on CI
+        if get(ENV, "BUILDKITE", "") == "true"
+            artifact_url = "artifact://$figpath"
+            print("\033]1338;url='$(artifact_url)';alt='$(alt)'\a\n")
+        end
     end
-    Plots.png(
-        Plots.plot(
-            [1:dt:T],
-            relL2err,
-            xlabel = "time [s]",
-            ylabel = "Relative L₂ err",
-            label = "",
-        ),
-        joinpath(path, "HeightRelL2errorVstime.png"),
-    )
-    linkfig(
-        relpath(
+    @info "Test case: $(test_name)"
+    @info "  with α: $(test_angle_name)"
+    @info "Solution L₂ norm at time t = 0: ", norm(Y0_global.h)
+    @info "Solution L₂ norm at time t = $(T): ", norm(sol_u_end_global.h)
+    @info "Fluid volume at time t = 0: ", sum(Y0_global.h)
+    @info "Fluid volume at time t = $(T): ", sum(sol_u_end_global.h)
+
+    if test_name == steady_state_test_name ||
+       test_name == steady_state_compact_test_name
+        # In these cases, we use the IC as the reference exact solution
+        @info "L₁ error at T = $(T): ", norm(solution[end].h .- Y0_global.h, 1)
+        @info "L₂ error at T = $(T): ", norm(solution[end].h .- Y0_global.h)
+        @info "L∞ error at T = $(T): ",
+        norm(solution[end].h .- Y0_global.h, Inf)
```

Comment:
Can we include a direct comparison of the serial/ parallel run solution `norm` outputs ?

Reply:
This is best done outside the driver, possibly in the CI pipeline if needed.

Reply:
I will add the corresponding distributed run in the CI pipeline for comparision.